src/gadgets/mpt_update.rs

```diff
@@ -83,7 +83,9 @@ impl<F: FieldExt> MptUpdateLookup<F> for MptUpdateConfig {
         let proof_type = self.proof_type.current() * is_start();
         let old_value = self.old_value.current() * is_start();
         let new_value = self.new_value.current() * is_start();
-        let address = self.intermediate_values[0].current() * is_start();
+        let [address_high, address_low, ..] = self.intermediate_values;
+        let address =
+            address_high.current() + address_low.current() * Query::Constant(F::from_u128(1 << 96));
```

Comment:
This is incorrect. Already fixed here: https://github.com/scroll-tech/mpt-circuit/commit/0bae9eeb813583c11f6db1f961a7e92f8c9bda82